services/src/main/java/org/keycloak/broker/saml/mappers/UserAttributeMapper.java

```diff
@@ -99,61 +101,84 @@ public String getDisplayType() {
     @Override
     public void preprocessFederatedIdentity(KeycloakSession session, RealmModel realm, IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
         String attribute = mapperModel.getConfig().get(USER_ATTRIBUTE);
-        String value = getAttribute(mapperModel, context);
-        if (value != null) {
-            if (attribute.equalsIgnoreCase("email")) {
-                context.setEmail(value);
-            } else if (attribute.equalsIgnoreCase("firstName")) {
-                context.setFirstName(value);
-            } else if (attribute.equalsIgnoreCase("lastName")) {
-                context.setLastName(value);
+        Map<String, List<String>> attributeToValues = getAttribute(mapperModel, context);
+        if (!attributeToValues.isEmpty()) {
+            String attributeName = getAttributeName(mapperModel);
+            if (attribute.equalsIgnoreCase(EMAIL)) {
+                setIfNotEmpty(context::setEmail, attributeToValues.get(attributeName));
```

Comment:
Will it work? `attributeToValues.get` returns `List<String>`, but context::setEmail expects plain `String` parameter